2021-01-28 11:46:15,196:urllib3.connectionpool:Starting new HTTPS connection (1): developers.symphonyfintech.in:443
2021-01-28 11:46:16,606:urllib3.connectionpool:https://developers.symphonyfintech.in:443 "POST /interactive/user/session HTTP/1.1" 200 None
2021-01-28 11:46:16,606:root:Main Login Successful
2021-01-28 11:46:16,606:root:Calculating everything before executing orders..
2021-01-28 11:46:16,606:root:Calculating weekly and monthly expiry dates..
2021-01-28 11:46:16,606:root:weekly expiry is : 28Jan2021, monthly expiry is: 28Jan2021
2021-01-28 11:46:16,722:root:('StrikePrice computed as : ', 13850)
2021-01-28 11:46:16,722:root:Input of get_eID fn : NIFTY, ce,28Jan2021,13850
2021-01-28 11:46:16,722:urllib3.connectionpool:Starting new HTTPS connection (1): developers.symphonyfintech.in:443
2021-01-28 11:46:17,779:urllib3.connectionpool:https://developers.symphonyfintech.in:443 "GET /marketdata/instruments/instrument/optionsymbol?exchangeSegment=2&series=OPTIDX&symbol=NIFTY&expiryDate=28Jan2021&optionType=CE&strikePrice=13850 HTTP/1.1" 200 None
2021-01-28 11:46:17,779:root:Exchange Instrument ID : 41438, {41438: 'CE'}
2021-01-28 11:46:17,779:root:Input of get_eID fn : NIFTY, pe,28Jan2021,13850
2021-01-28 11:46:17,779:urllib3.connectionpool:Starting new HTTPS connection (1): developers.symphonyfintech.in:443
2021-01-28 11:46:18,813:urllib3.connectionpool:https://developers.symphonyfintech.in:443 "GET /marketdata/instruments/instrument/optionsymbol?exchangeSegment=2&series=OPTIDX&symbol=NIFTY&expiryDate=28Jan2021&optionType=PE&strikePrice=13850 HTTP/1.1" 200 None
2021-01-28 11:46:18,828:root:Exchange Instrument ID : 41439, {41438: 'CE', 41439: 'PE'}
2021-01-28 11:46:18,828:root:Checking margin from user..
2021-01-28 11:46:18,828:root:Checking balance..
2021-01-28 11:46:18,835:urllib3.connectionpool:Starting new HTTPS connection (1): developers.symphonyfintech.in:443
2021-01-28 11:46:21,157:urllib3.connectionpool:https://developers.symphonyfintech.in:443 "GET /interactive/user/balance HTTP/1.1" 200 None
2021-01-28 11:46:21,157:root:Balance retreived success, available cash : 1000000000
2021-01-28 11:46:21,157:root: 
              net_cash - 1000000000,
              strikePrice - 13850,
              Nifty LTP - 13840.4,
              BankNiftyLTP - None
              exchangeInstrumentID - [41438, 41439]
              
2021-01-28 11:46:21,157:root:All set from prepareVars func.. Giving a go for orders..
2021-01-28 11:46:21,157:root:required variables set-- ready to trigger orders at desired time
2021-01-28 11:47:00,271:root:Required Margin Available.. Taking positions...
2021-01-28 11:47:01,229:urllib3.connectionpool:Starting new HTTPS connection (1): developers.symphonyfintech.in:443
2021-01-28 11:47:01,241:urllib3.connectionpool:Starting new HTTPS connection (1): developers.symphonyfintech.in:443
2021-01-28 11:47:01,261:urllib3.connectionpool:Starting new HTTPS connection (1): developers.symphonyfintech.in:443
2021-01-28 11:47:01,279:urllib3.connectionpool:Starting new HTTPS connection (1): developers.symphonyfintech.in:443
2021-01-28 11:47:02,129:urllib3.connectionpool:https://developers.symphonyfintech.in:443 "POST /interactive/user/session HTTP/1.1" 200 None
2021-01-28 11:47:02,144:root:Main Login Successful
2021-01-28 11:47:02,144:urllib3.connectionpool:https://developers.symphonyfintech.in:443 "POST /interactive/user/session HTTP/1.1" 200 None
2021-01-28 11:47:02,144:root:Main Login Successful
2021-01-28 11:47:02,144:root:Placing Orders with StopLoss..
2021-01-28 11:47:02,144:root:placing order for --, 41438
2021-01-28 11:47:02,160:urllib3.connectionpool:Starting new HTTPS connection (1): developers.symphonyfintech.in:443
2021-01-28 11:47:02,160:root:Placing Orders with StopLoss..
2021-01-28 11:47:02,160:root:placing order for --, 41439
2021-01-28 11:47:02,160:urllib3.connectionpool:Starting new HTTPS connection (1): developers.symphonyfintech.in:443
2021-01-28 11:47:02,182:urllib3.connectionpool:https://developers.symphonyfintech.in:443 "POST /interactive/user/session HTTP/1.1" 200 None
2021-01-28 11:47:02,182:root:Main Login Successful
2021-01-28 11:47:02,182:urllib3.connectionpool:https://developers.symphonyfintech.in:443 "POST /interactive/user/session HTTP/1.1" 200 None
2021-01-28 11:47:02,198:root:Main Login Successful
2021-01-28 11:47:02,984:urllib3.connectionpool:https://developers.symphonyfintech.in:443 "POST /interactive/orders HTTP/1.1" 400 87
2021-01-28 11:47:02,984:root:orderId and Traded price in try block of placeOrderWithSL: 0 and 0
2021-01-28 11:47:02,984:root:placing SL order for -- {symbol}
2021-01-28 11:47:02,984:root:TradedPice is not available, Hence not placing SL order, TRY MANUALLY..
2021-01-28 11:47:02,984:root:orderDictionary from placeOrderWithSL {41438: [41438, 0, 0]}
2021-01-28 11:47:03,031:urllib3.connectionpool:https://developers.symphonyfintech.in:443 "POST /interactive/orders HTTP/1.1" 400 87
2021-01-28 11:47:03,031:root:orderId and Traded price in try block of placeOrderWithSL: 0 and 0
2021-01-28 11:47:03,031:root:placing SL order for -- {symbol}
2021-01-28 11:47:03,031:root:TradedPice is not available, Hence not placing SL order, TRY MANUALLY..
2021-01-28 11:47:03,031:root:orderDictionary from placeOrderWithSL {41439: [41439, 0, 0]}
2021-01-28 11:47:03,031:root:new dict values :  {'oo': [], 'tt': [], 'qq': [], 'ss': [], 'sl': []}
2021-01-28 11:47:03,147:root:login initializing..
2021-01-28 11:47:03,147:urllib3.connectionpool:Starting new HTTPS connection (1): developers.symphonyfintech.in:443
2021-01-28 11:47:03,965:urllib3.connectionpool:https://developers.symphonyfintech.in:443 "POST /interactive/user/session HTTP/1.1" 200 None
2021-01-28 11:47:03,965:root:Login Successful
2021-01-28 11:47:03,965:urllib3.connectionpool:Starting new HTTPS connection (1): developers.symphonyfintech.in:443
2021-01-28 11:47:04,821:urllib3.connectionpool:https://developers.symphonyfintech.in:443 "POST /marketdata/instruments/subscription HTTP/1.1" 200 None
2021-01-28 11:47:04,836:root:Failed in PNL
Traceback (most recent call last):
  File "C:\Users\mling\anaconda3\lib\site-packages\pandas\core\indexes\base.py", line 2895, in get_loc
    return self._engine.get_loc(casted_key)
  File "pandas\_libs\index.pyx", line 70, in pandas._libs.index.IndexEngine.get_loc
  File "pandas\_libs\index.pyx", line 101, in pandas._libs.index.IndexEngine.get_loc
  File "pandas\_libs\hashtable_class_helper.pxi", line 1675, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas\_libs\hashtable_class_helper.pxi", line 1683, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'tradedPrice'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "F:\Downloads\Python\First_Choice_Git\xts\strategy\Strategy_1.py", line 230, in printPNL
    df['pnl']=(df['ltp']-df['tradedPrice'])*df['qty']
  File "C:\Users\mling\anaconda3\lib\site-packages\pandas\core\frame.py", line 2902, in __getitem__
    indexer = self.columns.get_loc(key)
  File "C:\Users\mling\anaconda3\lib\site-packages\pandas\core\indexes\base.py", line 2897, in get_loc
    raise KeyError(key) from err
KeyError: 'tradedPrice'
